### PR TITLE
[docker] Fix docker 'development' build failure

### PR DESCRIPTION
**Pull request by ConeyLiu**
## Why are these changes needed?
This patch fixes the docker development build failed due to permissions issue.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docker/ray-deps/Dockerfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ ENV LANG=C.UTF-8
 COPY $WHEEL_PATH .
 RUN $HOME/anaconda3/bin/pip --no-cache-dir install $(basename $WHEEL_PATH)[all] \
     $(if [ "$AUTOSCALER" = "autoscaler" ]; then echo \
+        "six==1.13.0" \
```
Comment (ijrsvt) [inline docker/ray-deps/Dockerfile:12]: Out of curiosity, what is this for?

File: docker/development/Dockerfile [hunk 2]
```diff
@@ -5,8 +5,8 @@ FROM rayproject/ray-deps:latest
 ADD ray.tar /ray
 ADD git-rev /ray/git-rev
 # Install dependencies needed to build ray
-RUN apt-get update && apt-get install -y curl unzip cmake gcc && apt-get clean
-RUN cd /ray && git init && ./ci/travis/install-bazel.sh --system
+RUN sudo apt-get update && sudo apt-get install -y curl unzip cmake gcc g++ && sudo apt-get clean
+RUN sudo chown -R ray /ray && cd /ray && git init && ./ci/travis/install-bazel.sh --system
```
Comment (ijrsvt) [inline docker/development/Dockerfile:9]:
```suggestion
RUN sudo chown -R ray:users /ray && cd /ray && git init && ./ci/travis/install-bazel.sh --system
```

**Review by ijrsvt — COMMENTED** [review]
Thank you so much for adding this! We really appreciate it! Two small comments :)

**Comment by ijrsvt** [comment]
Can you also update the description?

File: docker/ray-deps/Dockerfile (hunk 1 above)
Comment (ConeyLiu) [reply docker/ray-deps/Dockerfile:12]: Hi @ijrsvt, `google-api-core` requires `six >= 1.13.0`.

**Comment by ConeyLiu** [comment]
Hi @ijrsvt, thanks for the review. Addressed comments and updated the description.

**Comment by ConeyLiu** [comment]
cc @ijrsvt

**Review by ijrsvt — APPROVED** [review]
LGTM

**Pull request merged**